src/convertToObject.js

```diff
@@ -6,7 +6,26 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const styles = sourceString
+    .trim()
+    .split(';')
+    .filter((style) => style.trim() !== '');
+
+  const stylesObject = styles.reduce((acc, style) => {
+    if (!style.length || style.indexOf(':') === -1) {
+      return acc;
+    }
+
+    const [property, value] = style.split(':');
+
+    if (property.length > 0) {
+      Object.assign(acc, { [property.trim()]: value.trim() });
+    }
+
+    return acc;
+  }, {});
```

Comment:
The use of `reduce` is appropriate for transforming the array into an object. However, the condition inside the `reduce` callback is a bit redundant. The check for `style.length` is unnecessary because you've already filtered out empty strings. Additionally, you could directly destructure the property and value within the `split` method call to avoid trimming them separately later.